source/adios2/core/adios2_libinfo.cpp

```diff
@@ -0,0 +1,163 @@
+/*
+ * Distributed under the OSI-approved Apache License, Version 2.0.  See
+ * accompanying file Copyright.txt for details.
+ *
+ * adios2_libinfo.h
+ *
+ *  Created on: June 22, 2023
+ *      Author: Norbert Podhorszki pnorbert@ornl.gov
+ */
+
+#include "adios2_libinfo.h"
+#include "adios2/common/ADIOSConfig.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+const int adios2_version_major = ADIOS2_VERSION_MAJOR;
+const int adios2_version_minor = ADIOS2_VERSION_MINOR;
+const int adios2_version_patch = ADIOS2_VERSION_PATCH;
+const char *const adios2_version_str = ADIOS2_VERSION_STR;
+
+static const char *aae[] = {"BP3",
+                            "BP4",
+#ifdef ADIOS2_HAVE_BP5
+                            "BP5",
+#endif
+#ifdef ADIOS2_HAVE_HDF5
+                            "HDF5",
+#endif
+#ifdef ADIOS2_HAVE_SST
+                            "SST",
+#endif
+#ifdef ADIOS2_HAVE_MPI
+                            "SSC",
+#endif
+#ifdef ADIOS2_HAVE_DataMan
+                            "DataMan",
+#endif
+#ifdef ADIOS2_HAVE_DataSpaces
+                            "DataSpaces",
+#endif
+                            "Inline",
+#ifdef ADIOS2_HAVE_DAOS
+                            "DAOS",
+#endif
+#ifdef ADIOS2_HAVE_MHS
+                            "MHS",
+#endif
+#ifdef ADIOS2_HAVE_CATALYST
+                            "ParaViewADIOSInSituEngine",
+#endif
+                            "Null",
+                            "Skeleton",
+                            nullptr};
+
+void adios2_available_engines(int *nentries, char const *const **list)
```

Comment:
```suggestion
void adios2_available_engines(int *nentries, const char *const **list)
```

To match the style used in ADIOS2. AFAIK, const char and char const is the same. Same below